Sources/NIOFileSystem/Internal/SystemFileHandle.swift

```diff
@@ -1116,17 +1116,28 @@ extension SystemFileHandle {
         if delayMaterialization {
             // When opening in this mode we can more "atomically" create the file, that is, by not
             // leaving the user with a half written file should e.g. the system crash or throw an
-            // error while writing. On Linux we do this by opening the directory for the path
-            // with `O_TMPFILE` and creating a hard link when closing the file. On other platforms
-            // we generate a dot file with a randomised suffix name and rename it to the
+            // error while writing. On non-Android Linux we do this by opening the directory for
+            // the path with `O_TMPFILE` and creating a hard link when closing the file. On other
+            // platforms we generate a dot file with a randomised suffix name and rename it to the
             // destination.
+            #if os(Android)
+            return Self.syncOpenWithMaterialization(
```

Comment:
Rather than duping the whole call, can we conditionalise just `useTemporaryFileIfPossible`? i.e.

```swift
#if os(Android)
let useTemporaryFileIfPossible = false
#else
let useTemporaryFileIfPossible = true
#endif

Makes it a little bit easier to see what the differences are.
```

Reply:
Done